### PR TITLE
[prometheusremotewriteexporter] Translate resource to the target info metric

### LINKED ISSUE #8265: [prometheus components] Translate the "target" info metric to/from resource attributes
The spec was updated in https://github.com/open-telemetry/opentelemetry-specification/pull/2381 to require the prometheus receiver to convert the "target" info metric to resource attributes, and to recommend converting resource attributes to the "target" info metric when exporting.

See https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#supporting-target-metadata-in-both-push-based-and-pull-based-systems for a description of the "target" info metric.

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8265.

The only part that wasn't specified in the Spec is what to use as the timestamp of the target info metric.  I've chosen to use the most recent timestamp in the batch.

Alternatives: 
* We could set it to time.Now(), but then it would have a different timestamp from the rest of the batch, which might cause some strange behavior.
* We could also pick a random timestamp from the batch and use that.  That would work well in the standard case in which all scraped prometheus metrics have their timestamp set to the scrape time.

**Testing:**

Added unit tests

### CODE REVIEW

File: pkg/translator/prometheusremotewrite/go.mod [hunk 1]
```diff
@@ -15,6 +15,7 @@ require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect
 	github.com/kr/pretty v0.3.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.46.0 // indirect
```
Comment (Aneurysm9) [inline pkg/translator/prometheusremotewrite/go.mod:18]:
```suggestion
	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.47.0 // indirect
```

This should probably also have a `replace` directive for this module.

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@
 - `coralogixexporter` Allow exporter timeout to be configured (#7957)
 - `prometheusremotewriteexporter` support adding trace id and span id attached to exemplars (#8380)
 - `influxdbexporter`: accept histogram metric missing infinity bucket. (#8462)
+- `prometheusremotewriteexporter`: Translate resource attributes to the target info metric (#8493)
```
Comment (Aneurysm9) [inline CHANGELOG.md:29]: Make sure this goes in the `Unreleased` section.

Comment (dashpole) [reply CHANGELOG.md]: done

File: pkg/translator/prometheusremotewrite/go.mod (hunk 1 above)
Comment (dashpole) [reply pkg/translator/prometheusremotewrite/go.mod]: done

**Review by codeboten — COMMENTED** [review]
@Aneurysm9 please review, your comments have been addressed.

File: pkg/translator/prometheusremotewrite/go.mod [hunk 3]
```diff
@@ -15,8 +15,11 @@ require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect
 	github.com/kr/pretty v0.3.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.47.0 // indirect
```
Comment (Aneurysm9) [inline pkg/translator/prometheusremotewrite/go.mod:18]:
```suggestion
	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.48.0 // indirect
```

Apologies, this slid into yet another release window.  Approving pending this suggestion so hopefully that doesn't happen again.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**